File: chroma-league/src/main/java/net/booone/chromaleague/hud/animations/StaticPartialBlinkingAnimation.java
package net.booone.chromaleague.hud.animations;

import net.booone.chromaleague.hud.PredefinedKeySets;
import net.booone.razersdk.animation.AnimatedFrame;
import net.booone.razersdk.animation.SimpleFrame;
import net.booone.razersdk.color.Color;
import net.booone.razersdk.color.StaticColor;

public abstract class StaticPartialBlinkingAnimation extends AnimatedFrame {
    private static final int BLINK_TIMES = 8;

    protected StaticPartialBlinkingAnimation(Color color) {
        for (int i = 0; i < BLINK_TIMES; i++) {
            addAnimationFrame(new SimpleFrame(PredefinedKeySets.BLACKWIDOW_FUNCTIONAL, color));
            addAnimationFrame(3, new SimpleFrame(PredefinedKeySets.BLACKWIDOW_FUNCTIONAL, StaticColor.BLACK));
        }
    }
}

File: src/hud/animations/StaticPartialBlinkingAnimation.ts
import {AnimatedFrame} from '../../razer-sdk/animation/AnimatedFrame.js';
import {SimpleFrame} from '../../razer-sdk/animation/SimpleFrame.js';
import {Color} from '../../razer-sdk/color/Color.js';
import {StaticColor} from '../../razer-sdk/color/StaticColor.js';
import {BLACKWIDOW_FUNCTIONAL} from '../PredefinedKeySets.js';

const BLINK_TIMES = 8;

export class StaticPartialBlinkingAnimation extends AnimatedFrame {
    constructor(color: Color) {
        super();
        for (let i = 0; i < BLINK_TIMES; i++) {
            this.addAnimationFrame(new SimpleFrame(BLACKWIDOW_FUNCTIONAL, color));
            this.addAnimationFrame(3, new SimpleFrame(BLACKWIDOW_FUNCTIONAL, StaticColor.BLACK));
        }
    }
}
